web/src/drafts.js

```diff
@@ -167,6 +167,22 @@ export function rename_stream_recipient(old_stream_id, old_topic, new_stream_id,
     }
 }
 
+export function count_same_recipient_drafts(state) {
+    const current_drafts = draft_model.get();
+    let count = 0;
+    for (const draft_id of Object.keys(current_drafts)) {
+        const draft = current_drafts[draft_id];
+        if (
+            (state.type === "stream" && util.same_stream_and_topic(draft, state)) ||
+            (state.type === "private" &&
+                state.private_message_recipient === draft.private_message_recipient)
+        ) {
+            count = count + 1;
+        }
+    }
+    return count;
+}
+
 export function snapshot_message() {
     if (!compose_state.composing() || compose_state.message_content().length <= 2) {
```

Comment:
Are you missing live-update for when a new draft is created/deleted?

Reply:
A new draft is created only on closing the compose box, hence I didn't add live update for that, but for the case of deleting drafts, I'll add live update. Thanks for pointing this out!

Reply:
Added calls to the updating function both when adding and deleting a draft for consistency